pkg/cloud/services/network/gateways_test.go

```diff
@@ -149,3 +150,117 @@ func TestReconcileInternetGateways(t *testing.T) {
 		})
```

Comment:
Can we check the conditions set on AWSCluster in clusterscope object for success case?

Reply:
I think we could check this condition. But as per current PR scope, I have added the missing unit test cases and not focused on improving the existing test cases.
However to improve the existing test cases we could start a separate issue or can add it as part of adding integration tests(if required). Wdyt?